foreign.js

```diff
@@ -0,0 +1,23 @@
+const fs = require('fs');
+const path = require('path');
+const foreign = require('foreign');
+const readFile = (filename, callback) => fs.readFile(filename, {encoding: 'utf8'}, callback);
+const relative = (filename) => path.join(process.argv[2], filename);
+
+const main = () => {
+  const complete = (error, files) => {
+    error ?
+      process.stderr.write(error.message + '\n') :
+      process.stdout.write(files.join(''));
+
+    process.exit(error ? 1 : 0);
+  };
+
+  readFile(relative('index.txt'), (error, index) => {
+    if(error) return complete(error);
```

Comment:
I consider this misleading use of `return`. When using `return` for control flow only, one should make this clear by writing `return;` (rather than `return <expr>;`). Even better, we could avoid using `return` at all:

```javascript
if (err == null) {
  foreign.parallel(readFile, index.match(/^.*(?=\n)/gm).map(relative), complete);
} else {
  complete(err);
}
```